cv32e40s/tests/programs/custom/illegal_access_loop_tests/illegal_access_loop_tests.c

```diff
@@ -0,0 +1,89 @@
+/*
+**
+** Copyright 2022 OpenHW Group
+**
+** Licensed under the Solderpad Hardware Licence, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+**
+** https://solderpad.org/licenses/
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+**
+*******************************************************************************
+**
+** Contains generated tests on U-mode access and privilege instructions.
+**
+*******************************************************************************
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include "corev_uvmt.h"
+#include <stdint.h>
+
+// extern and global variable declaration
+extern volatile void  setup_pmp();
+extern volatile unsigned int illegal_full(), csr_loop(), mco_loop(), Check_mcounteren();
+
+
+// Declaration of assert 
+static __inline__ void assert_or_die(uint32_t actual, uint32_t expect, char *msg) {
+  if (actual != expect) {
+    printf(msg);
+    printf("expected = 0x%lx (%ld), got = 0x%lx (%ld)\n", expect, (int32_t)expect, actual, (int32_t)actual);
+    exit(EXIT_FAILURE);
+  }
+}
+
+void change_m_mode(void) {
+  //s4 is x20
+  __asm__ volatile("addi s4, x0, 0x3"); // add 3 to the s4 register, which will be read by the trap handler late
```

Comment:
`s4` is a preserved register, so the function that called this one might already be using it.
If we randomly change it here, then the rest of the program might get corrupted.

For communicating with a trap handler, I would rather recommend using global variables. `gbl_mysignaltothehandler`.

Reply:
I agree, I'll change this and update the PR today.